src/controllers/companyController.js

```diff
@@ -0,0 +1,87 @@
+import { companies } from "../models/index.js";
+import notFoundError from "../erros/notFoundError.js";
+
+class CompanyController {
+  static async registerCompany(req, res, next) {
+    try {
+      const newCompany = await companies.create(req.body);
+      res.status(201).json({ message: "Loja criada!", company: newCompany });
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  static async listCompanyById(req, res, next) {
+    try {
+      const id = req.params.id;
+      const foundCompany = await companies.findById(id);
+
+      if (foundCompany !== null) {
+        res.status(200).json(foundCompany);
+      } else {
+        next(new notFoundError("Id do livro não localizado"));
+      }
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  static async updateCompany(req, res, next) {
+    try {
+      const id = req.params.id;
+      const foundCompany = await companies.findById(id);
+
+      if (foundCompany !== null) {
+        const id = req.params.id;
+        const updatedCompany = await companies.findByIdAndUpdate(id, req.body, {
+          runValidators: true,
+          new: true,
+        });
+        res
+          .status(200)
+          .json({ message: "Loja atualizada!", company: updatedCompany });
+      } else {
+        next(new notFoundError("Id do livro não localizado"));
+      }
```

Comment:
Aqui fica uma dica. Tu não precisas buscar o documento antes de atualizá-lo, pois o próprio `findByIdAndUpdate` retorna null caso o documento não seja encontrado através do ID informado. Então tu pode resumir todo esse código, e tornar essa operação atômica:

```suggestion
       const id = req.params.id;
      const updatedCompany = await companies.findByIdAndUpdate(id, req.body, {
        new: true,
      });

      if (!updatedCompany) {
        next(new notFoundError("Company not found"));
        return;
      }

      res
        .status(200)
        .json({ message: "Loja atualizada!", company: updatedCompany });
```

Reply:
✅